<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="robot_name" default="MR1"/>
  <arg name="manager_name" default="nodelet_manager"/>
  <arg name="nodelet_mode" default="standalone"/>


  <group ns="$(arg robot_name)">

<!-- Nodelet Manager -->
    <group if="$(eval nodelet_mode=='load')">
      <node pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" />
    </group>

<!-- CAN -->
    <node pkg="nodelet" type="nodelet" name="slcan_bridge" 
    args="$(arg nodelet_mode) can_plugins/SlcanBridge $(arg manager_name)" >
      <param name="port" type="string" value="/dev/ttyACM0" />
    </node>

    <node pkg="nodelet" type="nodelet" name="can_handler" 
    args="$(arg nodelet_mode) can_plugins/CanHandler $(arg manager_name)" />

<!--Motor Driver-->
    <node pkg="nodelet" type="nodelet" name="motor1" args="$(arg nodelet_mode) can_plugins/Md201xNode $(arg manager_name)" >
      <param name="bid" type="string" value="640" />
      <remap from="motor1_cmd" to="launch1_cmd"/>
      <remap from="motor1_cmd_val" to="launch1_vel"/>
    </node>
    <node pkg="nodelet" type="nodelet" name="motor2" args="$(arg nodelet_mode) can_plugins/Md201xNode $(arg manager_name)" >
      <param name="bid" type="string" value="62c"/>
      <remap from="motor2_cmd" to="launch2_cmd"/>
      <remap from="motor2_cmd_val" to="launch2_vel"/>
    </node>
    <node pkg="nodelet" type="nodelet" name="motor3" args="$(arg nodelet_mode) can_plugins/Md201xNode $(arg manager_name)" >
      <param name="bid" type="string" value="63c" />
      <remap from="motor3_cmd" to="launch3_cmd"/>
      <remap from="motor3_cmd_val" to="launch3_vel"/>
    </node>
    
    
    
    <node pkg="nodelet" type="nodelet" name="MR1_nodelet_main" args="$(arg nodelet_mode) MR1/MR1_nodelet_main $(arg manager_name)" >
      
      <param name="Arrow_Pick_Arm_deg" type="double" value="2.5" />
      
    </node>
    
<!--Solenoid_Driver-->  
    <node pkg="solenoid_driver_node" type="solenoid_driver_node" name="solenoid" >
      <param name="bid" type="string" value="100" />
      <param name="name" value="solenoid"/>
      <remap from="solenoid_cmd" to="solenoid_cmd"/>
      <remap from="solenoid_order" to="solenoid_order"/>
    </node>
<!-- joy -->
    <rosparam file="$(find MR1)/param/logitech_f710_joy_assign.yaml" command="load" />
    <node respawn="true" pkg="joy" type="joy_node" name="joy_node" >
      <param name="dev" type="string" value="/dev/input/js0" />
      <param name="deadzone" value="0.12" />
    </node>
  </group>  
</launch>   
    
